### PR TITLE
docs: 1.6 upgrade docs & mitigation steps

### PR DESCRIPTION
**Pull request by laurens**
## What does this PR do?

Fix 1.6 upgrade docs with findings from #2273 

As parts of this are mitigations for a current hotfix, please advise or take-over which parts make sense to you to include in the official docs.

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/formbricks-com/app/docs/self-hosting/migration-guide/page.mdx [hunk 1]
```diff
@@ -29,12 +29,16 @@ To run all these steps, please navigate to the `formbricks` folder where your `d
 <CodeGroup title="Backup Postgres">
 
 ```bash
-docker exec formbricks-quickstart-postgres-1 pg_dump -U postgres -d formbricks > formbricks_pre_v1.6_$(date +%Y%m%d_%H%M%S).sql
+docker exec formbricks-quickstart-postgres-1 pg_dump -Fc -U postgres -d formbricks > formbricks_pre_v1.6_$(date +%Y%m%d_%H%M%S).dump
```
Comment (laurens) [inline apps/formbricks-com/app/docs/self-hosting/migration-guide/page.mdx:32]: A note on why this change:

In a restore scenario sql files can only be restored with `psql` and require an empty database.

Using `-Fc` for *custom format* allows to use `pg_restore` (see below) to restore in one take including cleaning the database.

File: apps/formbricks-com/app/docs/self-hosting/migration-guide/page.mdx [hunk 2]
```diff
@@ -81,9 +85,38 @@ The above command will migrate your data to the latest schema. This is a crucial
...
 
+#### Restoring the database after a failed upgrade
+
+Replace the path to the `.dump` file and run:
+
+<Col>
+<CodeGroup title="Restore the database">
+
+```bash
+docker exec -i formbricks-quickstart-postgres-1 pg_restore --clean -U postgres -v -d formbricks < formbricks_pre_v1.6_20240320_123522.dump
```
Comment (ShubhamPalriwala) [inline apps/formbricks-com/app/docs/self-hosting/migration-guide/page.mdx:96]: Since the timestamp in `formbricks_pre_v1.6_20240320_123522` can change for each user, can we rewrite this to mention that a file similar to ....

**Review by ShubhamPalriwala — COMMENTED** [review]
Hey @laurens, thanks a lot for the PR 😊 Apologies that you faced the issue :( Super glad to see you open a PR to help others who might run into the same. True Open Source energy 🏆 

Just one small change I think that we can do, apart from that, looks solid to me 🚀

**Comment by ShubhamPalriwala** [comment]
And can you also review & sign the Formbricks CLA please since this is your first PR? 

https://github.com/formbricks/formbricks/pull/2295#issuecomment-2009593486

**Comment by ShubhamPalriwala** [comment]
Hey @laurens, thanks a lot for making the changes! Do you think we should mention this as well somewhere:
https://github.com/formbricks/formbricks/issues/2273#issuecomment-2009529938

Since this helped you resolve the bug!

**Comment by laurens** [comment]
@ShubhamPalriwala CLA accepted, but it cannot finish as it requires disclosure of email addresses in commits.

Feel free to recycle by all means. No need for a mention.

Thank you for formbricks!

**Comment by ShubhamPalriwala** [comment]
Hey @laurens, thanks for trying that out 😄! We can also add you manually to the CLA, can you give us a written statement that you agree to the CLA here itself! 

We'd really like to get your PR merged for the efforts that you've put in if it's possible 🙏🏼.
Let us know if that's possible or we can open a new PR from our end with your changes!

**Comment by mattinannt** [comment]
@laurens thanks a lot the contribution :-) Since you wrote you accepted the CLA it's all fine 😊
